service/src/main/java/com/fecklessweasel/service/objectmodel/StoredFile.java

```diff
@@ -173,12 +191,16 @@ private static int addToDatabase(Connection sql,
                                       String title,
                                       String description,
                                       Date creationDate,
-                                      String tag) throws ServiceException {
+                                      String tag,
+                                      String extension) throws ServiceException {
         OMUtil.sqlCheck(sql);
         OMUtil.nullCheck(user);
         OMUtil.nullCheck(course);
         OMUtil.nullCheck(title);
         OMUtil.nullCheck(description);
+        OMUtil.nullCheck(tag);
```

Comment:
We allow a file not to be tagged, so nullChecking the tag may cause use problems